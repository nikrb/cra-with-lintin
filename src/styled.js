import styled from 'styled-components';

export const Title = styled.h1`
  font-size: 1.5rem;
  text-align: center;
  color: palevioletred;
`;

export const Wrapper = styled.section`
  padding: 4em;
  background: papayawhip;
`;
export const Input = styled.input`
  padding: 0.5em;
  margin: 0.5em;
  color: palevioletred;
  background: papayawhip;
  border: none;
  border-radius: 3px;
  border: 2px solid transparent;

  &:focus {
    // outline: 2px solid green;
    border: 2px solid green;
  }
`;

export const CoolStyledButton = styled.button`
  background-color: #bada55;
  border: none;
  border-radius: 5px;
  box-shadow: 1px 1px 2px gray;
  margin: 2rem;
  outline: none;
  padding: 0.5rem;
  position: relative;
  width: 250px;

  &:active {
    box-shadow: 0 0 transparent;
    top: 1px;
  }
`;

export const IntroText = styled.p`
  font-size: 2rem;
`;

export const TitleWithParameter = styled.h1`
  color: ${p => (p.danger ? 'red' : 'inherit')};
  font-size: 2rem;
`;
