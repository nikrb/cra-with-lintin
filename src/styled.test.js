import React from 'react';
import renderer from 'react-test-renderer';
import { CoolStyledButton } from './styled';

test('renders cool button', () => {
  const btn = renderer.create(
    <CoolStyledButton>Test</CoolStyledButton>
  ).toJSON();
  expect(btn).toMatchSnapshot();
});
