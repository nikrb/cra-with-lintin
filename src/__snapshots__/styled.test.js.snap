// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders cool button 1`] = `
<button
  className="sc-bxivhb cqlAFD"
>
  Test
</button>
`;
